### PR TITLE
Feat/matic pol

### LINKED ISSUE #14589: MATIC to POL
Change app to use `POL` for Polygon. Do proper migration

### PR DESCRIPTION
**Pull request by tomasklim**
## Description

- change matic -> pol
- migration of suite settings matic -> pol

## Related Issue

resolve #14589

## Screenshots:
Screenshot 2024-09-03 at 13 45 22

### CODE REVIEW

File: packages/connect/src/backend/Blockchain.ts [hunk 1]
```diff
@@ -40,6 +40,11 @@ const getNormalizedShortcut = (shortcut: string) => {
         return 'XRP';
     }
 
+    // Can be safely removed when both Polygon PoS Blockbooks return POL as shortcut
+    if (shortcut.toLowerCase() === 'pol') {
```
Comment (vytick) [inline packages/connect/src/backend/Blockchain.ts]: shouldnt this be the other way around? checking 'matic' and returning 'pol' ?

Comment (tomasklim) [reply packages/connect/src/backend/Blockchain.ts]: Yes, thats why i have todo few lines below

**Comment by jvaclavik** [comment]
> Screenshot 2024-09-03 at 13 45 22

btw you can use 
```tsx
<Warning.Button textWrap={false}>...</Warning.Button>
```
 to prevent text wrapping in Warning.

**Review by jvaclavik — APPROVED** [review]
DS-related stuff ✅

**Review by jvaclavik — APPROVED** [review]

File: packages/connect/src/backend/Blockchain.ts (hunk 1 above)
Comment (tomasklim) [reply packages/connect/src/backend/Blockchain.ts]: 3a09c0211db5b258279e59c6d6e4aab89d1f460a this is okay, right?

**Review by mroz22 — APPROVED** [review]
connect part looks ok

**Review by vytick — APPROVED** [review]
Mobile related stuff looks good 👍

File: packages/suite/src/storage/migrations/index.ts [hunk 2]
```diff
@@ -885,4 +885,110 @@ export const migrate: OnUpgradeFunc<SuiteDBSchema> = async (
...
+                walletSettings.enabledNetworks[indexOfMatic] = 'pol';
+            }
+
+            return walletSettings;
+        });
+
+        await updateAll(transaction, 'suiteSettings', suiteSettings => {
+            suiteSettings.evmSettings.confirmExplanationModalClosed.pol =
+                // @ts-expect-error
+                suiteSettings.evmSettings.confirmExplanationModalClosed?.matic;
```
Comment (marekrjpolak) [inline packages/suite/src/storage/migrations/index.ts]: Is optional chaining here (and below as well) necessary or just a precaution? Just asking, it's perfectly fine to leave it there.

File: packages/suite/src/storage/migrations/index.ts [hunk 3]
```diff
@@ -885,4 +885,110 @@ export const migrate: OnUpgradeFunc<SuiteDBSchema> = async (
...
+        if (maticBackendSettings) {
+            backendSettings.add(maticBackendSettings, 'pol');
+            // @ts-expect-error
+            backendSettings.delete('matic');
+        }
+
+        const tokenManagement = transaction.objectStore('tokenManagement');
+        const maticTokenManagementShow = await tokenManagement.get('matic-coin-show');
+        const maticTokenManagementHide = await tokenManagement.get('matic-coin-hide');
+        if (maticTokenManagementShow && maticTokenManagementHide) {
```
Comment (marekrjpolak) [inline packages/suite/src/storage/migrations/index.ts]: This is migrated only when both _show_ and _hide_ is present? Isn't it a bit safer to migrate them separately?

File: packages/suite/src/storage/migrations/index.ts [hunk 4]
```diff
@@ -885,4 +885,110 @@ export const migrate: OnUpgradeFunc<SuiteDBSchema> = async (
...
+            return discovery;
+        });
+
+        await updateAll(transaction, 'txs', tx => {
+            // @ts-expect-error
+            if (tx.tx.symbol === 'matic') {
+                return null;
+            }
+
+            return tx;
```
Comment (marekrjpolak) [inline packages/suite/src/storage/migrations/index.ts]: It's not necessary (but neither forbidden) to return anything from `updateAll` unless something change.

File: packages/suite/src/storage/migrations/index.ts [hunk 5]
```diff
@@ -885,4 +885,110 @@ export const migrate: OnUpgradeFunc<SuiteDBSchema> = async (
...
+    if (oldVersion < 47) {
+        // matic persisted settings
+
+        await updateAll(transaction, 'walletSettings', walletSettings => {
+            // @ts-expect-error
+            const indexOfMatic = walletSettings.enabledNetworks.indexOf('matic');
+            if (indexOfMatic !== -1) {
+                walletSettings.enabledNetworks[indexOfMatic] = 'pol';
+            }
+
```
Comment (marekrjpolak) [inline packages/suite/src/storage/migrations/index.ts:903]: `lastUsedFeeLevel` in `walletSettings` not migrated or deleted.

File: packages/suite/src/storage/migrations/index.ts [hunk 6]
```diff
@@ -885,4 +885,110 @@ export const migrate: OnUpgradeFunc<SuiteDBSchema> = async (
...
+        const historicRates = transaction.objectStore('historicRates');
+        const historicRatesKeys = await historicRates.getAllKeys();
+        const historicRatesKeysWithMatic = historicRatesKeys.filter(key => key.includes('matic'));
+
+        historicRatesKeysWithMatic.forEach(key => {
+            historicRates.delete(key);
+        });
+
+        const sendFormDrafts = transaction.objectStore('sendFormDrafts');
```
Comment (marekrjpolak) [inline packages/suite/src/storage/migrations/index.ts:1054]: `formDrafts` (probably trade section?) not migrated or deleted.

**Comment by bosomt** [comment]
**QA OK**

- migration - Matic account disappear, so you need to disable and enable coins /will be fixed/
- sent POL tx ✅ 
- sent token tx ✅ 
- links to blockbook ✅ 

tested on 24.8.3 az well
- changed https://matic1.trezor.io to https://pol1.trezor.io 
- sent token ✅ 
- sent pol ✅ 
- link to POL Ledger accounts leads to Bitcoin /works fine on 24.10.0

**Info:**
 - Suite version: desktop 24.10.0 (262a1ca8f00091e670a5377f084ac0eb1a396097)
 - Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) TrezorSuiteDev/24.10.0 Chrome/124.0.6367.243 Electron/30.3.1 Safari/537.36
 - OS: MacIntel
 - Screen: 1512x982
 - Device: Trezor T2T1 2.8.1 regular (revision 632b9561559b7ab6824bb7eeac072874e07b7b82)
 - Transport: BridgeTransport 2.0.33

**Comment by FreeWall** [comment]
`coingeckoNativeId` in [networksConfig.ts](https://github.com/trezor/trezor-suite/pull/14093/files#diff-8275801eb25562af454cd9d02084b668b58a396ff45c85242da60e8a203c46fdR416) should be `polygon-ecosystem-token`
https://www.coingecko.com/en/coins/polygon-ecosystem-token

File: packages/suite/src/storage/migrations/index.ts (hunk 3 above)
Comment (tomasklim) [reply packages/suite/src/storage/migrations/index.ts]: They should be both present at once. But sure, I will split it

File: packages/suite/src/storage/migrations/index.ts (hunk 5 above)
Comment (tomasklim) [reply packages/suite/src/storage/migrations/index.ts:903]: ah true

File: packages/suite/src/storage/migrations/index.ts (hunk 6 above)
Comment (tomasklim) [reply packages/suite/src/storage/migrations/index.ts:1054]: 🤯

File: packages/suite/src/storage/migrations/index.ts [hunk 7]
```diff
@@ -890,4 +890,207 @@ export const migrate: OnUpgradeFunc<SuiteDBSchema> = async (
...
+        });
+
+        await updateAll(transaction, 'accounts', account => {
+            // @ts-expect-error
+            if (account.symbol === 'matic') {
+                account = { ...account, symbol: 'pol' };
+            }
+
+            return account;
+        });
```
Comment (tomasklim) [inline packages/suite/src/storage/migrations/index.ts]: the migration stops here and does not continue, any idea why? @marekrjpolak

Comment (tomasklim) [reply packages/suite/src/storage/migrations/index.ts]: Screenshot 2024-09-26 at 21 57 11

Comment (marekrjpolak) [reply packages/suite/src/storage/migrations/index.ts]: Resolved in person. Long story short, `updateAll` util is not capable of updating entries including their primary key, so they have to be removed and re-added instead.

**Comment by marekrjpolak** [comment]
> the migration stops here and does not continue, any idea why?

The relevant part looks good now, so miniapprove from me :+1:

**Pull request merged**